### PR TITLE
feat: #100 rp adapter: present-proof with issuer

### LINKED ISSUE #100: rp adapter: send request-presentation to the issuer
part of #60

### PR DESCRIPTION
**Pull request by llorllale**
depends on #94

closes #100 

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

File: pkg/restapi/rp/controller_test.go [hunk 1]
```diff
@@ -68,3 +70,27 @@ func (m *mockAriesStorageProvider) TransientStorageProvider() ariesstorage.Provi
 
 	return ariesmockstorage.NewMockStoreProvider()
 }
+
+type stubPresentProofClient struct {
```
Comment (rolsonquadras) [inline pkg/restapi/rp/controller_test.go:74]: We can move this to common [`mock`](https://github.com/trustbloc/edge-adapter/tree/master/pkg/internal/mock/) folder.

File: pkg/restapi/rp/operation/operations.go [hunk 2]
```diff
@@ -404,6 +443,20 @@ func (o *Operation) peekInvitationData(id string) *invitationData {
 	return o.transientInvitationData[id]
 }
 
+func (o *Operation) setThidInvitationData(d *thidInvitationData) {
+	o.thidInvDataLock.Lock()
+	defer o.thidInvDataLock.Unlock()
+
+	o.thidInvitationDataMap[d.threadID] = d
```
Comment (rolsonquadras) [inline pkg/restapi/rp/operation/operations.go:450]: Shouldn't this be in some store instead of local map ?

Comment (llorllale) [reply pkg/restapi/rp/operation/operations.go:450]: @rolsonquadras it's on my TODO list: #29

File: pkg/restapi/rp/controller_test.go (hunk 1 above)
Comment (llorllale) [reply pkg/restapi/rp/controller_test.go]: @rolsonquadras done

**Pull request merged**